cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/utils/ConfigurationUtils.java

```diff
@@ -0,0 +1,240 @@
+     * folder in it
+     * @param customTemplatesLocation
+     *            File location to store in configuration file
+     * @throws IOException
+     *             if the configuration file could not created
+     */
+    public void createConfigFile(File customTemplatesLocation) throws IOException {
+        Path path = Paths.get(getCobigenCliRootPath() + File.separator + COBIGEN_CONFIG);
+        Properties props = new Properties();
+        props.setProperty(COBIGEN_CONFIG_LOCATION_KEY, customTemplatesLocation.toString());
+        props.store(new FileOutputStream(path.toFile()), MessagesConstants.CUSTOM_LOCATION_OPTION_DESCRIPTION);
+    }
+
+    /**
+     * Reads the configuration file and returns all of its properties
+     * @return Properties
+     */
+    public Properties readConfigFileProperties() {
+        Properties props = new Properties();
+
+        Path path = Paths.get(getCobigenCliRootPath() + File.separator + COBIGEN_CONFIG);
+        try (BufferedReader reader = Files.newBufferedReader(path, Charset.forName("UTF-8"))) {
+            props.load(reader);
+        } catch (IOException e) {
+            logger.error("An error occured while reading the config file", e);
+        }
+        return props;
+    }
+
+    /**
+     * Processes the input file's path. Strips the quotes from the file path if they are given.
+     * @param inputFile
+     *            the input file
+     * @return input file with processed path
+     */
+    public File preprocessInputFile(File inputFile) {
+        String path = inputFile.getPath();
+        String pattern = "[\\\"|\\'](.+)[\\\"|\\']";
+        boolean matches = path.matches(pattern);
+        if (matches) {
+            path = path.replace("\"", "");
+            path = path.replace("\'", "");
+            return new File(path);
+        }
+
+        return inputFile;
+    }
+
+    /**
+     * Tries to load a class over it's file path. If the path is /a/b/c/Some.class this method tries to load
+     * the following classes in this order: <list>
+     * <li>Some</li>
+     * <li>c.Some</li>
+     * <li>b.c.Some</li>
+     * <li>a.b.c.Some</> </list>
+     * @param classPath
+     *            the {@link Path} of the Class file
+     * @param cl
+     *            the used ClassLoader
+     * @return Class<?> of the class file
+     * @throws ClassNotFoundException
+     *             if no class could be found all the way up to the path root
+     */
+    public Class<?> loadClassByPath(Path classPath, ClassLoader cl) throws ClassNotFoundException {
+        // Get a list with all path segments, starting with the class name
+        Queue<String> pathSegments = new LinkedList<>();
+        // Split the path by the systems file separator and without the .class suffix
+        String[] pathSegmentsArray = classPath.toString().substring(0, classPath.toString().length() - 6)
+            .split("\\".equals(File.separator) ? "\\\\" : File.separator);
```

Comment:
Please use the Path API to do that, do not do this on your own.

Reply:
fixed in core already.